src/backend/empty/src/descriptor.rs

```diff
@@ -0,0 +1,42 @@
+use hal::pso;
+use log::debug;
+
+/// Dummy descriptor pool.
+#[derive(Debug)]
+pub struct DescriptorPool;
+
+impl pso::DescriptorPool<crate::Backend> for DescriptorPool {
+    unsafe fn allocate_set(
+        &mut self,
+        _layout: &DescriptorSetLayout,
+    ) -> Result<DescriptorSet, pso::AllocationError> {
+        Ok(DescriptorSet {
+            name: String::new(),
+        })
+    }
+
+    unsafe fn free_sets<I>(&mut self, descriptor_sets: I)
+    where
+        I: IntoIterator<Item = DescriptorSet>,
+    {
+        descriptor_sets.into_iter().for_each(std::mem::drop);
+    }
+
+    unsafe fn reset(&mut self) {
+        debug!("Resetting descriptor pool");
+    }
+}
+
+#[derive(Debug)]
+pub struct DescriptorSetLayout {
+    /// User-defined name for this descriptor set layout
+    pub(crate) name: String,
+    /// Bindings contained in this layout
+    pub(crate) bindings: Vec<pso::DescriptorSetLayoutBinding>,
```

Comment:
do we need these fields?

Reply:
I thought it would add some "weight" to the structure, to simulate the fact that a real implementation would also use _some_ memory to retain this kind of data.

Reply:
I've removed the `bindings` field, but I'll keep the name since it might be useful for debugging (through `set_descriptor_set_layout_name`) and also adds some "weight".